include/lomse_staffobjs_table.h

```diff
@@ -181,15 +181,6 @@ class ColStaffObjs
                 }
             }
 
-            virtual ~iterator() {}
```

Comment:
This change is not exactly needed to fix this warning but there doesn't seem to be a reason to inherit from this iterator class, and no class in Lomse does it. Hence there is no need for a virtual destructor here so I have removed it as well along with the explicit `operator=` definition.

Reply:
I started long ago with Lomse and by that time I had experience with C but little experience and knowledge of C++. I new about the `rule of three` and the improved version, the `rule of five`, much later but still I'm not used to take this rule into account when defining classes and also I never reviewed the code to comply with it. Probably you will see many definitions that need fixes. I usually define destructors as `virtual` without more considerations, just in case I derive for that class.

Also, another problem I have is in proper usage of `const` qualifier. Whenever I try to use it I find problems that force me to duplicate definitions with different `const` combinations, and, usually I opt by the shortest path: to remove the `const` qualifier and leave just one definition valid for all cases.

Please, fell free to fix these issues when you find any, without an specific PR for the fix.